improver/psychrometric_calculations/hail_size.py

```diff
@@ -391,17 +441,27 @@ def dry_adiabatic_descent_to_ccl(
         )
         return t_dry
 
-    def get_hail_size(self, vertical: np.ndarray, horizontal: np.ndarray) -> np.ndarray:
+    def get_hail_size(
+        self, vertical: np.ndarray, horizontal: np.ndarray, wet_bulb_zero: np.ndarray
+    ) -> np.ndarray:
         """Uses the lookup_table and the vertical and horizontal indexes calculated
-        to extract and store values from the lookup nomogram. Masked data points or
-        if vertical or horizontal values are negative lead to a hail_size of 0.
+        to extract and store values from the lookup nomogram.
+
+        The hail size will be set to 0 if
+            1) there are masked data points,
+            2) vertical or horizontal values are negative,
+            3) the wet bulb freezing altitude is greater that 4400m.
```

Comment:
Stephen needs to check this. I think it should be wet bulb freezing height (i.e. the distance between ground level and freezing level as we want the hail size when it reached ground level, not sea level). This would mean subtracting the orography height from the web bulb freezing altitude.

Reply:
Stephen has checked this - we should be using wet bulb freezing level height above ground level.

Reply:
I've updated the plugin to also take in the orography cube and subtract it from wet bulb height above sea level (and updated relelvant doc strings and tests).